primitives/runtime/src/generic/digest.rs

```diff
@@ -374,6 +392,42 @@ impl<'a, Hash> DigestItemRef<'a, Hash> {
 	pub fn try_to<T: Decode>(&self, id: OpaqueDigestItemId) -> Option<T> {
 		self.try_as_raw(id).and_then(|mut x| Decode::decode(&mut x).ok())
 	}
+
+	/// Try to match this to a `Self::Seal`, check `id` matches and decode it.
+	///
+	/// Returns `None` if this isn't a seal item, the `id` isn't of variant seal,
+	/// the `id` identifier doesn't match or when the decoding fails.
+	pub fn seal_try_to<T: Decode>(&self, id: OpaqueDigestItemId) -> Option<T> {
```

Comment:
Wouldn't it make more sense to just pass `ConsensusEngineId` as argument here? Otherwise people can do stupid things like `digest.seal_try_to(OpaqueDigestItemId::Consensus(...))`.

Reply:
Yeah, I was not sure either :D I can change it.